src/widgets/dialogs/UserInfoPopup.cpp

```diff
@@ -867,6 +882,23 @@ void UserInfoPopup::updateUserData()
     this->ui_.ignoreHighlights->setEnabled(false);
 }
 
+void UserInfoPopup::updateFocusLoss()
+{
+    this->closeAutomatically_ = !this->closeAutomatically_;
+    if (this->closeAutomatically_)
+    {
+        this->setActionOnFocusLoss(BaseWindow::Delete);
+        if (this->ui_.pinButton != nullptr)
+            this->ui_.pinButton->setPixmap(getApp()->themes->buttons.pin);
+    }
+    else
+    {
+        this->setActionOnFocusLoss(BaseWindow::Nothing);
+        if (this->ui_.pinButton != nullptr)
+            this->ui_.pinButton->setPixmap(getResources().buttons.pinEnabled);
+    }
+}
```

Comment:
`updateFocusLoss` should not be responsible for toggling the `this->closeAutomatically_` value, rather it should read it as you do on and act on it.

It must also be called in the constructor once to initiate the default value.